### ISSUE TITLE
Representing Multi Issuer Credentials in the VCDM

### ISSUE THREAD

**Issue by decentralgabe** [w3c/vc-data-model#932]
As per the discussion at TPAC on September 16:

# Overview

Multiple issuers: where there are multiple DID or URI represented issuers of a credential to a single (or multiple) parties.

This issue mostly lays out information. I'm interested in what the community thinks, and moving towards a concrete proposal (with suggestions from you).

## Related To

- https://github.com/w3c/vc-data-model/issues/931
- https://github.com/w3c/vc-data-model/issues/930

## Prior Art

- If you have some, please share.

## Use Cases

- NFTs from a smart contract (contract address vs minter address?)
- Co-tweets
- Parent's claims about their children
- Treaties, nuclear launch codes
- Co-signing for a loan

## What Would It Look Like?

<img width="810" alt="Screen Shot 2022-09-16 at 10 42 40 AM" src="https://user-images.githubusercontent.com/7622243/190700724-56260182-7ba1-49b0-b2b8-7d65decb0501.png">

## Questions
- Should credentials with multiple issuers even be allowed?
- What alternatives are there?
- How can this be represented in JWTs, Data Integrity Proofs?
- Do existing selective disclosure/ZKP schemes support this construction?

**Comment by decentralgabe** [comment]
Mentioned this [in the issue around multiple subjects](https://github.com/w3c/vc-data-model/issues/931#issuecomment-1253081320), but had some more thoughts after TPAC to address the approach described last Friday:

One point that @smithsamuelm raised at TPAC (though relating to multiple issuers, not multiple subjects) was that a single DID can be used to represent multiple parties, if those parties agree to a representation scheme under a single DID (e.g. a multisig, multiple controllers, or something else). I believe this is *a possible* method of cleverly allowing multiple subjects (or issuers!) in a credential; however I am worried that this approach has two clear downsides:

1. It is not immediately clear in reading a credential who the subject(s) are. Many DIDs may have related keys/DIDs in them controlled by a single entity. It would be exceedingly difficult to determine cases of multiple subjects, or subjects who just have a bunch of identities.
2. DIDs with multiple keys/related IDs can change. This can harm the verifiability/integrity of a credential. For example, if I issued a credential to a single DID (`did:example:alices-multisig`) comprised of two parties (lets say `did:example:alice` and `did:example:bob`) and 6 months post-issuance Alice reforms her multisig to include new parties and remove Bob lets say `did:example:alices-multisig` now contains `did:example:alice`, `did:example:charlie`, and `did:example:dave`) is the credential to still be seen as valid?

Moving past that example. I'd like to gain a sense of folk's appetite for firming up the existing language around multiple subjects and introducing normative statements for the representation of multiple subjects in a single credential, expanding on [the existing spec text and example here](https://www.w3.org/TR/vc-data-model/#example-specifying-multiple-subjects-in-a-verifiable-credential).

**Comment by RieksJ** [comment]
This is not just technically adding things, but much more about its meaning. I would like to think that an issuer-signature over credential-data can be used to identify the [party](https://essif-lab.github.io/framework/docs/essifLab-glossary#party) on whose behalf an [actor](https://essif-lab.github.io/framework/docs/essifLab-glossary#actor) has signed the credential. Not just so that a verifier can question (perhaps also: sue) this party if so needed. There should be no more than one party, because a verifier would not be able to decide which of them would provide the authoritative answer to such questions (or would be suable). 

In a use-case where it would be beneficial for all members of a [community](https://essif-lab.github.io/framework/docs/essifLab-glossary#) to all sign a credential, I would say that the community (which is a party) should sign it. 

In  a use-case where an arbitrary set of parties would sign it (perhaps a petition), I would say that each of them sign their own copy of a credential that states the same.

I can imagine that the signature of different parties might be part of a credential, but NOT as an issuer-signature. An example is provided in the [decentralized governance paper](https://docs.google.com/document/d/1FQTxzQ9z9Tv-WA_UYyfF8AgvEfBYBWRgGvSdjsQof4s/edit#heading=h.vdf397laf9tu), that shows how parties can be certified/accredited to issue certain kinds of credentials, and embed this certificate in the VCs that they issue, enabling others to verify that the issuer has been certified. 

Let's not keep extending VCDM with all sorts of technical features that might bring (slight) benefits (at the technical level) and huge problems at the business/information/actual usage level.

**Comment by David-Chadwick** [comment]
We have many examples today of multiple signatures on documents, both concurrent signing and sequential signing. An example of the former is a cheque (US check) that needs multiple signatures. An example of the latter is when a witness signs a document after the contractual party has signed the document. So we should cater for both of these in the VC DM. In the NGI Atlantic project we are working on this problem in the context of JWT proofed VCs. Serial signing does not require any changes to the JWT signing algorithm, since the JWT output from the first signing can be used as the payload for the second signing. This process can be recursed as often as required. So I think it would be quite easy to describe this in the DM. For concurrent signing we propose to use JSON Web Signature JSON Serialization from RFC 7515 as we do not believe that the compact serialisation is required for such a document.

**Comment by OR13** [comment]
```
{
      "payload":
       "eyJpc3MiOiJqb2UiLA0KICJleHAiOjEzMDA4MTkzODAsDQogImh0dHA6Ly9leGF
        tcGxlLmNvbS9pc19yb290Ijp0cnVlfQ",
      "signatures":[
       {"protected":"eyJhbGciOiJSUzI1NiJ9",
        "header":  {"kid":"2010-12-29"},
        "signature":
         "cC4hiUPoj9Eetdgtv3hF80EGrhuB__dzERat0XF9g2VtQgr9PJbu3XOiZj5RZ
          mh7AAuHIm4Bh-0Qc_lF5YKt_O8W2Fp5jujGbds9uJdbF9CUAr7t1dnZcAcQjb
          KBYNX4BAynRFdiuB--f_nZLgrnbyTyWzO75vRK5h6xBArLIARNPvkSjtQBMHl
          b1L07Qe7K0GarZRmB_eSN9383LcOLn6_dO--xi12jzDwusC-eOkHWEsqtFZES
          c6BfI7noOPqvhJ1phCnvWh6IeYI2w9QOYEUipUTI8np6LbgGY9Fs98rqVt5AX
          LIhWkWywlVmtVrBp0igcN_IoypGlUPQGe77Rw"},
       {"protected":"eyJhbGciOiJFUzI1NiJ9",
        "header": {"kid":"e9bc097a-ce51-4036-9562-d2ade882db0d"},
        "signature":
         "DtEhU3ljbEg8L38VWAfUAqOyKAM6-Xx-F4GawxaepmXFCgfTjDxw5djxLa8IS
          lSApmWQxfKTUJqPP3-Kg6NU1Q"}]
     }
```

https://www.rfc-editor.org/rfc/rfc7515#appendix-A.6.4

Notice the missing `alg`... Not sure how much actual interop there is here... cc @selfissued @Sakurann @dwaite 

Is JWS multisignature well supported today? What are some pitfalls? Is there a compact serialization?

**Comment by OR13** [comment]
Here is part of the answer from the test vectors RFC: https://www.rfc-editor.org/rfc/rfc7520#section-4.8

**Comment by SmithSamuelM** [comment]
@David-Chadwick  Recursive Serial Signing works for endorsements where a later signer signs the signed payload from a previous signer, but it is problematic for thresholded multi-sig where the signing must not be ordered and the signers do not sign each others signed payload.

**Comment by decentralgabe** [comment]
Would it be reasonable to add normative spec language describing how to construct a multi-issuer credential and leave 'securing' to the Data Integrity and VC-JWT specs? It sounds like there are solutions for both camps.

**Comment by RieksJ** [comment]
> > There should be no more than one party, because a verifier would not be able to decide which of them would provide the authoritative answer to such questions (or would be suable).
> 
> This is similar to the confusion that has come up with holder/subject/prover. The answer from us must be clearer language to express possibilities, such as:
> 
> To be valid a verifier must receive authorization from:
> 
> 1. one of the issuers
> 2. a quorum of issuers
> 3. all issuers

+1 to the need for clearer language.

-1 to expressing additional possibilities without first having proper criteria for determining what does (not) qualify as a holder/subject/prover/issuer/verifier/validator/..., at least for use within the context of VCDM, which (for me) implies that all (active) stakeholders for the VCDM have demonstrated they have the same understanding of such terms.

**Comment by SmithSamuelM** [comment]
This is where I find cognitive dissonance with the VCDM based on Triples.   The exchange of VC is an information exchange. It is a transaction. Transactions have their own properties. The roles of the exchangers does not necessarily have anything to do with the information being exchanged. Or at the very least there is a narrow set of information that is needed to support the exchange of OTHER information as payload to the exchange that may be perfectly opaque to the exchange process itself.  When the other information and the information needed to support the exchange are confused then we have this cognitive dissonance.  This does not mean that a link between the exchange supporting information and the information in the exchange’s payload is ruled out. But only in some exchanges and only by virtue of shared identifiers. The same identifier can play a role a primary identifier to the exchange and also be referenced in the payload for some other role that is independent of the role in the exchange.  This makes it easy to define roles. The confusion of the terms holder/subject/issuer/… is a result of confusing and roles that support an exchange protocol with the roles of some higher level payload.

**Comment by SmithSamuelM** [comment]
What is the VC proof proving? Is it proof of issuance? Is it proof of Authorization by an issuer to and issuee. Is the proof related to a type of disclosure such as selective disclosure.  Is is a range proof?  Is is a herd privacy proof. The type of proof is entirely transactional. And that's my point. The VC data model is confused because proofs are part of transactions/interations between a prover and a provee not a distributed database in the sky that is semantic. Proofs are highly contextual because of they require a relationship between the parties engaged in the proof.

**Comment by SmithSamuelM** [comment]
If we only allowed on type of proof, that is proof of issuance with only an Issuer and no Issuee, then we could have a non-interactive non-transactional proof. But the majority of use cases for Credentials (which by defintion of credential is evidence of entitlement, means that there is an Entitler and Entitlee i.e. a transactional relationship and thats where the semantic web runs aground because it is not meant to be transactional. We need an lower layer that is transactional where all the proofs and associated transactions lay and the IDs related to the parties engaged in those transactions with roles defined by the transaction and then a higher layer that is an opaque payload. Then we have clean separation of concerns and all the ambiguity we have been discussing for the last 4 years goes away. We will have extreme clarity.

**Comment by SmithSamuelM** [comment]
As a caveat, lest my use of the term semantic devolve into a discussion of what a semantic is.  To be clear transactions themselves have semantics, they have models. But the models are narrowly defined and constrained by the transaction. They are not the open world semantic model that the semantic web in general is trying to realize. And that is the cognative dissonance. Trying to stuff an open world data model into the very narrow closed world of A4 transactions.

**Comment by SmithSamuelM** [comment]
It seems to me that the VCWG2 is an opportunity to fix this unfortunate confusion of concerns and clearly delineate athe narrow semantics of an exchange layer from the broad semantics of a payload of the exchange layer. Maybe this splits the WG into two WGs once for doing all the A4, Confidentiality, Privacy exchange modeling and the other doing the payload modeling.

**Comment by SmithSamuelM** [comment]
So for example.  If I have only the need to provide a proof of issuance, then an authenticated key pair linked to an Issuer identifier that may or may not be linked to a natural persor or legal entity along with a non-repudiable digital signature made with that key pair provides a conceptually clean model for proof of issuance.  But as soon as I attempt to make that data for which I can prove issuance into an entitlement then I have two parties. The entitler and the entitlee. Each with a keypair and identifier with possibly links to natural persons and legal entities. Now I have to answer the question, what role does the entitlee play, in a given transaction is it the holder, the issuee, the discloser in a presentation of the credential to some other party. Oh wait what is the other party now. We have three parties. Could we have more parties. Yes of course. We could have a credential issued by an Entitler to and Entitlee that is held by some other party and that other party could actually be the discloser fo the credential to yet another party. How is this disclosing party related to the entitlee. Is is a proxy. is is a custodian, is it and agent? These are all transactional roles that have little to do with the payload of the entitlement itself and everything todo with the type of transaction involved in the exchange of the entitlement.  So we its not clear what a holder is, then it is clear that we have a confusion of concerns between the transactional exchange layer where the roles are clearly defined by the type of transaction and the payload layer that may or may not reference data related to the parties engaged in the transaction but should in all cases be opaque to the semantics of the transaction other wise its not a payload. This is how layered protocols are meant to work.

**Comment by SmithSamuelM** [comment]
A set of transactions that are exchanges can be used to build a complex model of relationships. One transactions can carry a payload of data that can later be referenced in support of some other transaction. The later transaction has clear semantics of the parties and roles to the transaction as defined by the transaction type. But either party may reference payloads of prior transactions in order to make decisions about a later transaction. This is how reputation works.  The temptation is to flatten the layers, and that is a temptation that should be resisted.  Collapsing layers increases complexity not decreases complexity. Complex systems are in general best understood as hierarchical compositions of information that functions best at a given layer in that hierarchy. There are applications where flattening layers can reduce complexity, but this is not one of them.

**Comment by David-Chadwick** [comment]
@SmithSamuelM "So for example. If I have only the need to provide a proof of issuance,"
But what if I have the need to provide proof of issuance to a specific entity? Then in that case the issuer would include the public key of the entity in the issued credential. This is still a clean model. There are still only two parties involved. The issuer and the issuee. 
But what if I need to provide proof of issuance that the credential is about a specific entity. Then in that case the issuer would include the public key of the credential subject. This is still a clean model as only two parties are involved.
And if I need to provide proof of issuance to a specific entity that the credential is about a third party, then the credential would include the public keys of the subject, the issuer and the issuee.
I see nothing wrong with having all of these requirements modelled in the VC DM v2. They are all independent of what happens next to the credential after it has been issued.

**Comment by TallTed** [comment]
@SmithSamuelM — It's hard to digest and respond properly to your walls of text, especially with incorrect punctuation (such as [many periods where question marks belong](https://github.com/w3c/vc-data-model/issues/932#issuecomment-1266088687)) and frequent typos (such as [instances of `is is` that should be `is it`](https://github.com/w3c/vc-data-model/issues/932#issuecomment-1266088687)).

I expect these challenges are much greater for the many folks working on this for whom English is a second or later language.

Please take some time to read over your last several comments, and correct as much as possible. Breaking up your larger streams of text into more consumable paragraphs will also help a lot. I'm sure that every minute you spend doing this will save (at least!) several minutes for your readers, certainly in aggregate and likely for each individual reader, who otherwise have to each spend time trying to figure out what you meant through the typos, etc.

**Comment by David-Chadwick** [comment]
CONCRETE PROPOSAL

Add the new role of _issuee_ to the V2 data model, which the issuer can optionally insert into the issued VC. 

The structure of _issuee_ is alligned to be the same as issuer (i.e. object or string).

**Comment by SmithSamuelM** [comment]
@TallTed Sorry for the typos. I will be more careful.

@David-Chadwick I believe that this is precisely the problem of ambiguity between holder/subject/... discussed above. Is the Issuee identifier counted as meta-data?  If the identifier of a party to the transaction is not meta-data then how does it fit into the VCDM. Clearly, any identifier can be represented as an object/string as metadata, but how does that object/string identifier fit into the subject/object/predicate model of the VCDM. We could represent every party associated with a transaction as metadata, and that would be just fine. Indeed, that is effectively what I am proposing. Because then this set of metadata can be used in its own transaction layer that supports a higher non-metadata layer.

**Comment by David-Chadwick** [comment]
@SmithSamuelM The reason for introducing the new role of issuee is to remove all ambiguity between subject and holder.
So, 
1. the issuer issues the VC to the issuee (always) and may optionally insert the latter's identity into the VC.
2. The issuer inserts its own identity into the VC it issues (always)
3. The issuer issues the VC about a subject (always) and may of may not uniquely identify the subject.

Once the VC has been issued it may pass from the issuee to no-one, or to the subject, or to a third party. Whoever it passes to (including no-one) is the holder of the VC until it is passed on again. It could of course be duplicated and passed to multiple holders. The holder is never identified in a VC because the VC is persistent and the holder is not.

**Comment by TallTed** [comment]
> [@David-Chadwick] 3. The issuer issues the VC about a subject (always) and may of [sic] may not uniquely identify the subject

(First, note the typo, "`may of`" should be "`may or`".)

Why change your phrasing for this, instead of following the same form as the first two, i.e., "`3. The issuer issues the VC about a subject (always) and may optionally insert the latter's identity into the VC`"?

(I would suggest also changing both instances of `the latter's` to be explicit, i.e., `the issuee's` in (1) and `the subject's` in (3), in your framing.  I would also suggest being explicit with `about one or more subjects`, as leaving this out may take us down another rabbit hole where "subject must be singular!")

**Comment by David-Chadwick** [comment]
Thanks Ted for correcting my typos. So how about:

1.   The issuer issues the VC to the issuee (always) and may optionally insert the issuee's identity into the VC.
2.   The issuer inserts its own identity into the VC it issues (always)
3.   The issuer issues the VC about one or more subjects (always) and may optionally insert the subject's (or subjects') identity into the VC.

**Comment by TallTed** [comment]
I suppose we should also support the possibility of multiple issuees (think, carbon copies) for a single VC...

And we should *always* differentiate between "identity" and "identifier".

1. The issuer always inserts its own identifier into any VC it issues.

2. The issuer always issues a VC to one or more issuees, and optionally inserts the issuee identifier(s) into the VC.

3. The issuer always issues a VC about one or more subjects, and optionally inserts the subject identifier(s) into the VC.

**Comment by decentralgabe** [comment]
@David-Chadwick I like your suggestion of issuee, I can see it clearing up confusion. Related to #931 it could be "issuees" if applicable as you and @TallTed both mention.

So, now we have issuer and issuers, and issuee and issuees. I'd like to get a PR going to encapsulate these changes but could use some assistance. Is anyone willing to pair with me to get a well-thought out change drafted?

**Comment by David-Chadwick** [comment]
I was going to produce a PR is my suggestion was positively received. So happy to work with you on this

**Comment by SmithSamuelM** [comment]
@TallTed  +1  
@David-Chadwick  +1

> CONCRETE PROPOSAL

+1 to Issuee(s)

**Comment by Sakurann** [comment]
(probably a topic for another coming issue), but -1 to a term `issuee` (not necessarily the concept). from a non-native speaker perspective, the term is not exactly intuitive and confusing.

**Comment by msporny** [comment]
Also -1 to `issuee` -- mostly because this overlaps heavily with the "holder binding" work that @awoie is doing and it doesn't sound like the people suggesting this proposal have talked to the people working on the "holder binding" work. Don't spend your time on a PR yet as it'll probably be -1'd unless it has high alignment w/ the "holder binding" work that @awoie is leading.

**Comment by David-Chadwick** [comment]
Holder binding is conceptually different to issuee binding because the role of holder is conceptually different to the role of issuee. The former is not fixed as holders can change from day to day. The issuee never changes, ever. So the issuer can bind a VC to an issuee, but never to "the holder" as the holder is whoever it happens to be at any point in time. So a subset of the techniques of the holder binding work can be used for issuee binding

**Comment by dwaite** [comment]
My concern is that a single property will not be robust enough. In particular, issuees tend to have their own properties - including the relationships with the subject(s) and issuer(s?), and any additional information needed for them to do a presentation proof. In some cases, the issuer may even try to convey whether or not that party is expected to issue statements about another party having a relationship to the original subject(s) or credential.

Since a credential can have multiple statements about different subjects, would it instead make sense to describe an issuee as another subject in a single credential, or possibly via a second credential?

**Comment by TallTed** [comment]
@msporny @dwaite @David-Chadwick @Sakurann — Please take your latest `issuee`-related comments to #942 (and hide them here) as this #932 is about multiple `issuer`s, not about `issuee`.  I should have done the same, sooner.  :-/

**Comment by RieksJ** [comment]
@SmithSamuelM wrote

> This is where I find cognitive dissonance with the VCDM based on Triples. The exchange of VC is an information exchange. 

I disagree. The exchange of VC is a data exchange, where the data represents information of the author, according to the mapping (semantics) that the author uses for that. No two mappings/semantics are the same. 

So here is an ambiguity issue related to the meaning of a signature. My understanding of a VC-signature is that the author says: these triples represent some knowledge that I have and that I consider to be true and trustworthy. If there are two signatures that have this meaning, it would imply that both parties would claim to have the identical knowledge, which, if you would argue that could happen, is very dangerous as it is difficult, if not impossible for third parties to establish.

Signatures on checques, contracts etc. have a different meaning. They state a commitment of the signer to abide by, support, etc. the contents of the data that it signs, in the meaning of the author of that document. And of course we need those. But they differ from a VC signature

**Comment by SmithSamuelM** [comment]
I am not sure I capture the meaningful difference between "an information exchange" and a "data exchange that represents information". ;)

In a layered protocol, the purpose of each layer is to isolate and encapsulate the information essential to the function of that layer. This enables the separation of concerns and avoids the leakage of information. The latter is an essential property of security (authenticity, confidentiality, and privacy) functionality. In a protocol stack there is a hierarchy of layers, where lower layers support higher layers. The higher layer may have visibility down into lower layers, but lower layers do not or should not have visibility into higher layers (the latter would break the encapsulation and leak information).  So finally, once the data has climbed the protocol stack, one could then after the fact, flatten the layers (because the topmost layer may have visibility into all the lower layers) and represent the flattened information is one construct, such as a bag of triples.  But I strongly suggest that doing so a priori and assuming that such a flattening does away with the need for a protocol stack of encapsulated layers in the first place is a dangerous equivocation.

**Comment by SmithSamuelM** [comment]
> So here is an ambiguity issue related to the meaning of a signature. My understanding of a VC-signature is that the author says: these triples represent some knowledge that I have and that I consider to be true and trustworthy. If there are two signatures that have this meaning, it would imply that both parties would claim to have the identical knowledge, which, if you would argue that could happen, is very dangerous as it is difficult, if not impossible for third parties to establish.

Exactly why we should not be mixing the layers, which your example does.

**Comment by SmithSamuelM** [comment]
Each Issuer should be empowered to define what a proof-of-issuance attributable to that Issuer means for a given issuance. This must be entirely contextualized to that issuance. The data model of the protocol stack, must then have a mechanism that enables the Issuer to express that definition in an interoperable way that enables verifiers to correctly evaluate the proof of issuance.    Multiiple signatures applied one issuance could mean something entirely different when applied to another issuance.    Consequently, IMHO, this means that each VC must have a type and that type is constrained by the Issuer to mean what the Issuer intends it to mean. 

In ACDC type is schema. Schema are immutable, Schema are identified by a universally unique content addressable identifier (a hash of the schema). This allows a permissionless registry of schema which define ACDC types. This enables the issuer to publish the rules of the road so to speak with regards the intepretation of what a proof-of-issuance conveys to any validator, without ambiguity.  By providing such a mechanism, the ACDC spec avoids having to define universal semantics only universal syntactical elements.

**Comment by SmithSamuelM** [comment]
So far with ACDC we only have identified the need for four roles. All the use cases we can think of are covered by those four roles because the typing system enables the semantics of the roles as used by a given ACDC to be adapted to the use case.

The four roles are:

ACDC Issuer
ACDC Issuee (the Issuee is optional and reflects a target identifier of an issuance. Not all issuances need a target identifier)

In the disclosure exchange protocol for an ACDC there are two roles

Discloser
Disclosee  (The disclosee is optional, a public broadcast disclosure may not benefit from a targeted disclosee identifier)

For multiple Issuers there are two mechanisms.

1) The Issuer identifier is controlled by a multi-sig thresholded group of identifiers. Each member of the group contributes one or more signing key pairs to the muti-sig group.  Proof-of-issuance requires a threshold satisfycing set of signatures from the multi-sig group.

2) Graph of independent ACDCs that are chained together to represent a group issuance as a tree. The origin of the tree is an ACDC that links back to the other independent ACDCs. The ACDCs independent ACDCs are the same type with the same attribute values (including Issuee) but each have a different issuer. A verifier can see that all the independent issuers made a verifiable commitment to the same attributes and Issuee. The origin ACDC in the graph is Issued by the Issuee of the other independent ACDCs. This Issuer to Issuee to Issuer chain provides a verifiable set of multiple Issuers using a graph. 

These two mechanisms mean we don't have a need to define a single ACDC with a list of Issuers. All ACDCs can have but one Issuer identifier but we can still securely convey the semantics and business logic of any multiple issuer situation.

**Comment by RieksJ** [comment]
Reading through this thread increasingly resembles me walking into a swamp: lots of good points are being made, but I have no clue as to their cohesion. There's a topic on (the meaning) of proofs/signatures - very valid, should be clarified (e.g. as in pk-certs). There's a topic on transactions (exchanging messages the meaning of which is to a considerable extent implied by the transaction context) - we need this as it links to business applications. There's a topic on issuees (and other 'roles' (whatever that means)).

It seems as if we're all playing in a field, but it isn't clear where the field begins, ends, and what the game actually is about. My suggestion is to sit back and think a bit better about what we're actually trying to do here.

My current thoughts on demarcating the playing field is that a VC is meta-data, claims, and proofs/signatures (P/S's), where:
- all claims originate from (the knowledge of) a single party, and there is one signature over the entire VC that has been created on behalf of that party that attests to this (which I call 'issuer' in this post). The semantics (meaning) of the claims originate from that party, who therefore must somehow publish this semantics, or refer to some published semantics (e.g.: a scheme). VCDM will specify what claims may look like in terms of structure/syntax, but any semantics are outside the scope of VCDM (as the semantics is (to be) controlled by the issuer of the VC.
- any other P/S can either be part of a claim, in which case they are out of scope of the VCDM (and must be specified by the issuer of that claim), or of the VC itself, in which case it is in scope of the VCDM, and its meaning must be explicitly specified. I support the argument that a particular (semantic) kind of P/S be only taken in consideration for inclusion in the VCDM if the purpose that it serves cannot be realized in another realistic fashion, e.g. by including it as part of a claim.  What constitutes 'realistic' is up for debate; it may include operational/performance arguments, as well as business/policy/governance arguments.
- similarly, any meta-data should be (re)considered along these lines of reasoning. It would make, e.g., an issuingdate part of the meta-data, but expiration date would be disputable. And in the end, it's what the majority of us agree on.

**Comment by theblockstalk** [comment]
We have built a multi-signature and delegated signature Verification module for verifiable credentials. This supports composable multi-party signatures through a combination of conditional signature Validation requirements expressed in a DID Document using the Conditional Proof verification method.

For this use case, you could create a new DAO-like account on the decentralised data registry, with a DID Document that expresses the need for e.g. 6 out of 8 other DIDs representing countries ( through a delegated account link ) to need to sign to create a valid signature.

We have just finished creating an implementation that works with the did-jwt(-vc) libraries and is DID, VC and JWT standards compliant.

Please read this blog article for information on how this works.
https://blog.tonomy.foundation/verifiable-credentials-with-provable-delegated-and-multi-sig-signatures-e46ca74d7d87

**Comment by decentralgabe** [comment]
I'd like to propose a PR that captures the following:
- Multi-issuer credentials are supported by the VCDM
- Securing the data model is a _separate_ concern from the language in the VCDM
- There can be multiple methods of securing multi-issuer credentials, such as: 
  - multisig schemes (multiple signatures combined into a single signature), 
  - wrapping credentials in credentials
  - chaining credentials (i.e. you issued x so now I can issue y)
  - multiple signatures over a single VC
  - a single issuer DID controlled by multiple parties

Please 👍 👎 if you would be supportive of such a PR.

**Comment by Sakurann** [comment]
+1 as long as it is clear that "There can be multiple methods of securing multi-issuer credentials" and each method will be defined in each securing spec

**Comment by msporny** [comment]
> I'd like to propose a PR that captures the following:

-0.7 on a PR that captures all of that, but only due to concerns regarding ecosystem complexity, more below:

> * Multi-issuer credentials are supported by the VCDM

The makes the ecosystem more complicated for a limited return on that complexity. It's not clear that the extra use cases justify the added complexity.

> * Securing the data model is a _separate_ concern from the language in the VCDM

+1 to the above.

> * There can be multiple methods of securing multi-issuer credentials, such as:

Yes, and this is where the concern comes in. The more options we have here, the more implementation complexity there will be. The question is: Can we achieve this multi-issuer use case by using the primitives we have today, such as a collection of VCs that say the same thing issued by different issuers?

>  * multisig schemes (multiple signatures combined into a single signature)

We could achieve the use case using a single issuer w/ a multisig scheme... that would re-use primitives we have today w/o adding the complexity of updating the `issuer` field.

>  * wrapping credentials in credentials

We could achieve the use case with a single issuer field today by taking this approach.

>  * chaining credentials (i.e. you issued x so now I can issue y)

We could achieve the use case with a single issuer field today by adding the 'content addressing other VCs' proposal that @dmitrizagidulin and @longpd put together.

>  * multiple signatures over a single VC

We could achieve the use case with a single issuer field today by doing this.

>  * a single issuer DID controlled by multiple parties

We could achieve the use case with a single issuer field today by doing this.

All this to say, we could keep the issuer field as a single value and do all of the things above to add issuers... OR, we could expand the issuer field to reference multiple issuers BUT, if we do that, we really should settle on (ideally) one of the mechanisms above. If we don't do that, it feels like we're begging for interop failures... or massive ecosystem complexity. I don't think it would be an easy ask of our engineering team to implement 6 different security schemes for the "multiple issuers" use case.

If we can focus on one mechanism, that would be a much easier ask (but I understand that it'll take work to get there).

**Comment by peacekeeper** [comment]
I don't really have an opinion on this, but regarding:

> * a single issuer DID controlled by multiple parties

I wanted to point people to this proposed verification method, which I think is really interesting:
https://w3c-ccg.github.io/verifiable-conditions/

**Comment by decentralgabe** [comment]
@msporny @dlongley I think (correct me if I'm wrong) I find your posts self-contradictory. You are saying _yes_ we do support multi issuer credentials, but with what we have today. And what we have today is not multi issuer credentials - meaning they're not allowed at all. I'd like to distinguish here between the concept of "multiple issuers" and "there are multiple values for the `issuer` property". It seems we agree on the former but not quite the latter yet.

Manu, you're correct that _many_ but not all multi-issuer schemes can leverage a single issuer field and that's a good thing, and an argument in favor of updating language to allow for multi-issuer credentials. The first spec text change would be to say _multi issuer credentials are allowed_ and one way (and maybe the only one that reaches consensus) of doing this is using the security mechanisms we have today.

I would recommend my first PR **does** update the issuer field to allow multiple values because there are cases where multiple issuers can be represented by a single signature, as in the case of multi-sig [schemes like schnorr](https://eprint.iacr.org/2018/068).

I would next recommend adding language about the possibility of supporting multiple signatures on a single credential, but this will be more controversial.

So, two PRs, possibly three for easier consensus (1 & 2 can be combined):
1. Multiple issuers are allowed with a single issuer value
2. Multiple issuers are allowed with multiple issuer values
----------
3. Multiple issuer are allowed with a variety of signature constructs (as listed above)

**Comment by dwaite** [comment]
For schemes where multiple issuers are actually multiple parties operating as a single _conceptual_ issuer, I would expect this single conceptual issuer to be indicated at the data model level, with further details within the proofing mechanism on which issuers had contributed to the proof and how. Having such multi-issuer support would be out of scope of the VCDM, but might affect other in-scope proofing mechanisms.

However, for multiple independent issuers to be represented at the data model layer, I would expect these to be multiple parties making the exact same statement with the exact same conditions - e.g. they are both asserting every predicate of the subject equally, are requiring the same expiry times, utilizing the same evidence, and so on. 

_Assuming_ a proof mechanism that meets the needs of an issuer which is a composite of multiple parties, is there a use case which mandates these constituents be represented as multiple issuers at the credential level, rather than a single composite issuer and a proof that explains who and how the credential was secured?

I would assume having this information within the proofing mechanism is far more appropriate, as it already would need to represent multiple independent proofs or describe parameters needed by the threshold scheme, associated with those constituent issuers. Likewise, I would assume the many permutations being represented by a common conceptual issuer would be needed, as someone needs to give common recommendations for verifiers on what is sufficient for verifiability and validation.

**Comment by SmithSamuelM** [comment]
@dwaite  +1

**Comment by decentralgabe** [comment]
@dwaite 

> Assuming a proof mechanism that meets the needs of an issuer which is a composite of multiple parties, is there a use case which mandates these constituents be represented as multiple issuers at the credential level, rather than a single composite issuer and a proof that explains who and how the credential was secured?

Yes, such as analogous real world credentials like marriage licenses, birth certificates, and treaties. It does not make sense to jam our concept of DID into the real world and say "well just go get a joint DID" because that's not what the VCDM is about. The VCDM intends to represent credentials, and there are numerous real world cases where multiple parties are issuers of the same credential. This is a truism about credentials that I don't believe should be up for debate. 

Again, I believe _securing_ the data model is a separate concern than representing statements about credentials in the data model itself, of which, having multiple issuers is one such statement.

**Comment by dwaite** [comment]
> Yes, such as analogous real world credentials like marriage licenses, birth certificates, and treaties. It does not make sense to jam our concept of DID into the real world and say "well just go get a joint DID" because that's not what the VCDM is about. The VCDM intends to represent credentials, and there are numerous real world cases where multiple parties are issuers of the same credential. This is a truism about credentials that I don't believe should be up for debate.

Would this not imply a much larger change than just turning issuer into a set?  

I would not expect from your scenarios that it would only a matter of having multiple issuer identities (and supporting proof infrastructure). I would expect that each issuer could attach their own conditions for validity onto the credential. For instance, how would a single `credentialStatus` work with multiple issuers?

The verifier does need to still understand credential validity and lifecycle, and the validity rules for high value credentials is hopefully not ad-hoc. Multi-party issuance requires even more coordination, and such policy needs to be understood for the validity of credentials to be quantified. This is why I am trying to understand if this can be captured within the proof section, even if that means a single issuer 'identifier' is then referring to more of a process or framework than to a specific party. And no, it doesn't need to be a DID.

However, I also suspect some of these scenarios do not require multi-party representation. For example - if the witnesses for a marriage license are unable to serve any official capacity such as changing my taxation structure, imparting new legal rights, or revoking said license, were they actually an issuer? Or did they provide a statement (possibly in the form of a verifiable credential) to the more broadly accepted issuing authority, and were potentially captured as such within evidence or as a predicate?

**Comment by decentralgabe** [comment]
> how would a single credentialStatus work with multiple issuers?

it would make sense to define a credentialStatus property that has multi-issuer support. that's what extensibility is all about!

> The verifier does need to still understand credential validity and lifecycle, and the validity rules for high value credentials is hopefully not ad-hoc

correct, as they do today in the real world with multi-issuer credentials

Re your marriage example: I don't believe the witnesses are the issuers, the county recorder's office is (at least in the US). Witnesses attestations would be considered evidence like you suggest. I was referring to the couple to be wed's signatures on the application for a marriage license as a multi-issuer credential, sorry I did not make this clear.

I'm certain we _can_ find ways around this, but I wonder if that's our job to tell the world how credentials ought to be represented, instead of being flexible to the many real world credentials and behaviors that exist today.

**Comment by dlongley** [comment]
@decentralgabe,

I think the problem here is that it seems insufficient to just modify the `issuer` property to accept a set of values instead of a single value in order to address the use cases. Given that its sufficiency to address the use cases hasn't yet been demonstrated, it doesn't yet make sense to change the data model. Changing it before we know if it works can lead to churn, broken implementations, and more problems for us to solve.

Another way of putting this is: yes, there is some separation between securing the data model and the data model itself, but they are connected. Every use case requires securing the data model. So, if we change the data model to address a use case from a modeling perspective, but in so doing render it too difficult to secure, then we will just fail the use case from a security perspective.

Additionally, there are other ways of addressing the use cases that have been mentioned and are being explored. I think there's rough consensus that the use cases are valid. Where we don't yet have rough consensus is on the mechanism we ought to use to address them.

> I'm certain we can find ways around this, but I wonder if that's our job to tell the world how credentials ought to be represented, instead of being flexible to the many real world credentials and behaviors that exist today.

I agree we shouldn't be dismissive of the use cases. However, we should question whether a sort of [skeuomorphic approach](https://marco.org/2010/03/11/overdoing-the-interface-metaphor) to this problem is appropriate, whereby we try to mirror exactly what's going on with physical credentials today vs. just ensuring the technology does indeed address the use case.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-02-01-vcwg#section4-1) on 2023-02-01

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 4.1. Representing Multi Issuer Credentials in the VCDM (issue vc-data-model#932)

_See github issue [vc-data-model#932](https://github.com/w3c/vc-data-model/issues/932)._




**Kristina Yasuda:** Issue 932.. Can anyone talk about this specific issue about multi-issue credentials?.  

**Manu Sporny:** At a high level, this is about whether a VC can have multiple issuers..  
… It's a bit deeper than it sounds. On the data model level, of course we could have an array. But then it's not clear what you do with the proofs..  
… Perhaps we want the issuer to be a single field (like it is today), and then have multiple signatures.  
… You may have one primary issuers but 'n' signatures on the credential..  
… And then there's the argument that we shouldn't complicate the ecosystem. When you have multiple issuers, ultimately there needs to be an issuing authority..  
… E.g. you have multiple schools in a district that test the students, but you only have one authority that issues the credentials..  
… There are different aspects on what this is about, and Gabe is leading the discussion..  

> *Dave Longley:* or just issue two (or N-many) VCs ... need clear use cases and how those could / ought to be solved.

**Kristina Yasuda:** We discussed mechanisms such as chaining, I think GS1 has something in production with multiple issuers..  
… Do we want to define how the multiple issue scenario would work, and if yes which mechanisms do we want to use..  

**Gabe Cohen:** I see signing as a separate concern as the data model. It's important to clarify the data model first, if there can be multiple signatures..  
… A multi signature scheme could be used, or having multiple proofs. I think there is still work to be done in crypto suites..  
… I think the requirement is solid, I believe there are situations where single credentials are signed by multiple parties, and the spec text should clarify this..  
… We can add language to the spec, even if a crypt suite doesn't exist yet..  

> *Kristina Yasuda:* orie: +1 Gabe.

**Dmitri Zagidulin:** I agree that multiple issuers is a legitimate use case..  
… We have a similar situation in the DID data model, we have the notion of multiple controllers in a DID document.  
… We haven't done much with this mechanism,.  
… I recommed that we do not model multiple issuers as multiple signatures.  
… Either change the model to allow multiple issuers, or specify the notion of a compound issuer.  
… If a student has a diploma from 2 universities, there would be 1 issuer, which is the partnership/combination of the 2 universities..  

**Kevin Dean:** The GS1 example is not one of multiple issuers in this sense.  

> *Dave Longley:* +1 to the notion that there are multiple different ways this problem could be solved and needs further exploration.

**Kevin Dean:** The authority to issue a credential is derived from a prior credential. There is a chain of credentials, but this is not the same as having a credential with multiple issuers..  
… My authority to issue a final credential in a chain is derived from previous credentials in a chain..  
… Going back to private key cryptography in environments where parties don't trust each other..  

> *Kristina Yasuda:* orie: +1 Kevin, it's hierarchical.

**Kevin Dean:** Control of issuing a final credential is put either into completely unrelated party, or it's under some partial control of one of the parties. This might not be acceptable to others..  
… Environment where issuers don't necessarily trust each other, but they want to issue a credential together..  

> *Steve McCown:* +1 Kevin.

> *Dmitri Zagidulin:* treaties and contracts would be best modeled as a COLLECTION of VCs, from individual issuers.

**Kevin Dean:** For treaties, certain contracts, there may well be requirements for multple issuers in low trust environments..  

> *David Chadwick:* +1 kevin.

**Orie Steele:** I agree with what KevinDeanGS1 said and what decentralgabe said initially.  
… We have seen several security approaches to this problem.  

> *Dave Longley:* +1 to dmitri ... that it may be best to model these things as collections of VCs.

**Orie Steele:** Going back to what decentralgabe said, the data model is about the intention, and securing it is a separate concern..  
… Maybe you're used to OIDC or access tokens, where you have only 1 issuer and 1 subject..  
… The core data model should express the actual information that's important.  
… I feel uneasy how we talk about proofs in relation to the core data model..  
… I have a feel we perceive Data Integrity Proofs to be more "imporant" than JWS proofs.  
… The data model should be expressed in JSON-LD, and then the security aspect should be modeled using the proofs.  
… I'm aware of some of the work dmitriz mention regarding multiple controllers.  
… This is not equivalent to a data model that allows expressing multiple issuers.  

> *Kristina Yasuda:* what is the value of two issuers URIs if there is only one signature?.

**Manu Sporny:** The current data model restricts the "issuer" to be a single value. If we just talk about the data model, do we want to loosen that requirement?.  

> *Dave Longley:* multiple VCs, each signed by a different issuer, could all make the same claim about a credential subject (such as a treaty) ... and merging those together would represent all parties of the treaty signing together, etc. -- options to explore..

**Dmitri Zagidulin:** I think there would be great value if we restrict a VC to a single issuer..  
… In the education space, there is a "Comprehensive Learner Record", which is a kitchen sink approach that contains a lot of information.  

> *Kristina Yasuda:* orie: +1 Manu, it would be a change to the core data model first.

**Dmitri Zagidulin:** This can be a single "outer" credential with "nested" credentials. Each one has a single issuer. The "outer" one is signed by an aggregator..  

> *Andres Uribe:* +1 to dimitri.

> *Kerri Lemoie:* CLR spec: [https://1edtech.github.io/ComprehensiveLearnerRecord/docs/clr_v2p0.html](https://1edtech.github.io/ComprehensiveLearnerRecord/docs/clr_v2p0.html).

**Dmitri Zagidulin:** I think we can express these situations while keeping the restriction of having a single "issuer" in the data model.  

**Kristina Yasuda:** I don't think we're ready for a PR yet.  

> *Dmitri Zagidulin:* +1 dlongley, re having the /treaty/ itself be the subject of VCs.

**Kristina Yasuda:** I encourage continued conversation about the data model aspect of this..  

> *Dmitri Zagidulin:* that's the approach taken by our recent RWoT paper.

**Kristina Yasuda:** Thank you everyone, see you next week..  

---


</details>

**Comment by decentralgabe** [comment]
@dlongley

> Given that its sufficiency to address the use cases hasn't yet been demonstrated, it doesn't yet make sense to change the data model. Changing it before we know if it works can lead to churn, broken implementations, and more problems for us to solve.

Yes, it should be really clear why making multiple issuers is a sufficient change. I raised [these points earlier](https://github.com/w3c/vc-data-model/issues/932#issuecomment-1253084616_), but let me try to expand.

## Option 1: Single credential, multiple issuers per single issuer identifier

A single conceptual issuer fails with the following considerations:
1. It does not immediately communicate to the holder or recipient(s) of VCs that there are multiple issuers. It requires additional resolution and parsing infrastructure which increases the burden for a credential to be understood.
2. Not all DID methods support multiple parties to be listed. This would be eliminating a broad class of users for multi-issuer credentials.
3. Not all DID methods support historical resolution of DID state. This means that there could be a point-in-time where the DID contained multiple issuers, and that point-in-time could be lost forever. This would make it impossible to know _who actually issued_ the credential and does not at all help further the multi-issuer use case.

Some of these points are present even without multi-issuer use cases, but we should not double down on these shortcomings, and instead create a new pathway to make it abundantly clear that a credential is multi-issuer and clearly communicate the provenance of a credential's issuance.

## Option 2: Multiple credentials, single issuer per credential

This approach fails the requirements for a number of real world use cases by increasing complexity and management overhead. We would need a way to normatively link credentials together, that is, to say that this credential is part of a multi-issuer pair. Complex, and confusing for both holders and verifiers--all interactions with these credentials now become a coordination problem.

There are two buckets of use cases that we have certainty we need this functionality for:

1. Where you need multiple signatures to authorize the same legal doc / authorization to take action  (e.g., [Unanimous Withness Consent](https://www.wonder.legal/us/modele/action-unanimous-written-consent#:~:text=An%20Action%20by%20Unanimous%20Written,face%2Dto%2Dface%20meeting.), Treaties, DAO actions, application for a marriage license, to adopt a child, or any joint resolution).
2. Where you need multiple signatories, but on different parts of a legal doc / different actions authorized by different signatories to unlock some common thing (e.g. signature by an individual and a notary for a home loan or title).

-----------------------------------------------------------

> So, if we change the data model to address a use case from a modeling perspective, but in so doing render it too difficult to secure, then we will just fail the use case from a security perspective. We can't give people half of a solution.

Agree, 100%. I think it's a chicken and egg scenario, and I'd advocate for changing the data model first because it is what everything else is built on top of. I do not think we should end up in a scenario where we block changes in the data model because higher layers have not yet demonstrated how they could work with the changes—this leads us to stalemate. We need to be able to innovate here first. 

Additionally, we already have indications of [how multiple signatures on a single credential can work, at least in some signature representations](https://github.com/w3c/vc-data-model/issues/932#issuecomment-1340604183).

> Additionally, there are other ways of addressing the use cases that have been mentioned and are being explored.

I believe these other ways are insufficient as I've outlined above. They cause confusion, have risk of changing from underneath you, and undermine what it means to be "verifiable."

**Comment by dlongley** [comment]
@decentralgabe,

> We would need a way to normatively link credentials together, that is, to say that this credential is part of a multi-issuer pair.

Why?

**Comment by decentralgabe** [comment]
> @decentralgabe,
> 
> > We would need a way to normatively link credentials together, that is, to say that this credential is part of a multi-issuer pair.
> 
> Why?

Without a standard way to do this you lose the value of multi-issuer creds. It's important to know there are multiple issuers from the credential, so that you can check their respective status lists, expirations, terms of use, and so on as may be required in some of these cases.

**Comment by dlongley** [comment]
@decentralgabe,

> Without a standard way to do this you lose the value of multi-issuer creds. It's important to know there are multiple issuers from the credential, so that you can check their respective status lists, expirations, terms of use, and so on as may be required in some of these cases.

I feel that we need to pull this back to first principles as I'm concerned that we keep hitting an erroneous conclusion. This is what I feel like I keep hearing:

1. There are use cases that need some mechanism whereby multiple parties make the same or similar assertions. (***Checks out to me***).
2. Those use cases could be solved using "multi-issuer credentials". (***I don't know, maybe***).
3. If we don't have a way to do X with "multi-issuer credentials", then "multi-issuer credentials" won't have value. (***If you say so***).
4. Therefore, we can't solve the use cases without X. (***I think this is an error in reasoning***)

The point I keep trying to make is that we should look at one or more specific use cases to see how we could solve them rather than jumping to "we must use multi-issuer credentials to solve the use cases we know are out there and matter, so let's talk about the best way to do that". I'm not convinced that "multi-issuer credentials" is the best way to the solve the use cases yet -- because it doesn't seem like we've fully explored the various ways to do it.

For example, you were just mentioning that there may be cases where N-many parties all need to make the assertion that a particular contract is valid. Well, that could be modeled by having each party issue a VC that says that said contract is valid. A verifier must *somehow* have a list of parties that they are looking to check to see if they've signed off on the contract -- and they can just run down that list and compare it against the set of VCs.

Now, this isn't a full-blown use case exploration, just a real quick example of how it seems like it could work off of the top of my head. It didn't need "multi-issuer credentials" (for some definition of that). My point is that we need to *really deeply explore a particular use case* to see what the possible solutions are. And, ideally, we repeat that for a few more use cases to see what's in common. Otherwise, it seems to me like we're caught in a logic loop that that is begging the question over whether "multi-issuer credentials" are the only way to solve the use cases.

**Comment by decentralgabe** [comment]
I'd like a response to the options and their critiques [I laid out above](https://github.com/w3c/vc-data-model/issues/932#issuecomment-1421175523), but I will say we should separate _can_ and _should_.

Perhaps many use cases of VCs could be done with centralized solutions (mDL anyone?) or even paper-based solutions as they have been for hundreds of years. This does not mean there isn't a _better way_ to do them with VCs or even multi-issuer VCs. Contorting the data model to "work" for these use cases won't get us anywhere until we agree that there are benefits in using a single credential with multiple issuers.

I agree with you that it's good to go deep on a single use case to start, so let's talk about a Unanimous Witness Consent document and apply the reasoning I linked above.

**Comment by dlongley** [comment]
@decentralgabe,

You asked me to respond to the options above -- though I do think it's working backwards as I mentioned. I will leave some comments on the options as you asked, but I really do recommend that people look into solving specific use cases first instead of looking at possible problems with possible solutions in the abstract.

> ## Option 1: Single credential, multiple issuers per single issuer identifier
> A single conceptual issuer fails with the following considerations:
> 
> 1. It does not immediately communicate to the holder or recipient(s) of VCs that there are multiple issuers. It requires additional resolution and parsing infrastructure which increases the burden for a credential to be understood.

Why does this need to be communicated at all? Maybe this is a feature, not a bug? This is why specific use cases need to be explored.

> 2. Not all DID methods support multiple parties to be listed. This would be eliminating a broad class of users for multi-issuer credentials.

You don't have to support this through DID methods. Additionally, mechanisms could be developed that abstract away the details of using any particular key registered with *any* DID method, including ones that produce static DID docs like `did:key`. A simple example of this is putting the private key in an HSM that is accessible via WebKMS where access to sign with the key is managed via ZCAPs. More complex examples could be constructed. This is just another example of not exposing the details to verifiers, thereby enabling more existing software to work without any changes.

> 3. Not all DID methods support historical resolution of DID state. This means that there could be a point-in-time where the DID contained multiple issuers, and that point-in-time could be lost forever. This would make it impossible to know _who actually issued_ the credential and does not at all help further the multi-issuer use case.

This seems orthogonal. Indeed, even if multiple issuers were listed in the same credential, I can imagine ways in which this concern would be unaffected.

> ## Option 2: Multiple credentials, single issuer per credential
> This approach fails the requirements for a number of real world use cases by increasing complexity and management overhead. We would need a way to normatively link credentials together, that is, to say that this credential is part of a multi-issuer pair. Complex, and confusing for both holders and verifiers--all interactions with these credentials now become a coordination problem.

We covered this above.

> There are two buckets of use cases that we have certainty we need this functionality for:
> 
> 1. Where you need multiple signatures to authorize the same legal doc / authorization to take action  (e.g., [Unanimous Withness Consent](https://www.wonder.legal/us/modele/action-unanimous-written-consent#:~:text=An%20Action%20by%20Unanimous%20Written,face%2Dto%2Dface%20meeting.), Treaties, DAO actions, application for a marriage license, to adopt a child, or any joint resolution).

It seems like the common link here is another agreed upon document that each party could reference in the VC they generated. That same document could list all of the expected signatories. You recommended that this be the particular use case to start chasing down in detail and it's a perfectly good choice.

> 2. Where you need multiple signatories, but on different parts of a legal doc / different actions authorized by different signatories to unlock some common thing (e.g. signature by an individual and a notary for a home loan or title).

It again seems like different issuers could issue different VCs referencing their sign off on the appropriate parts. In fact, it seems like this would get far more complicated / not be modeled properly if we used "multiple issuers" of the same VC here, but that's just a gut reaction. Again, a good reason that the use cases need to be rigorously explored.

Hopefully this is the response you were looking for. I recommend that people deep dive into the use case you suggested as a way to move the conversation forward around what potential solutions might look like.

**Comment by agropper** [comment]
Here’s a use-case:
https://www.politico.com/newsletters/digital-future-daily/2023/02/09/the-smallest-state-has-the-biggest-blockchain-ambitions-00082118

“…. opening a restaurant in Rhode Island currently requires accessing 11
state websites.“

Consider that each of the 11 websites constitutes a request on the part of
the restaurant. Consider that most of the sites would require an authorized
and accountable human to make the each request. Most, if not all, would
include:
- the accountable human
- their claims of authority
- the scope of their request (liquor, hours, etc…)
- the reason for the scope of the request
- payment

Also consider how likely it would be for the restaurant to want to
outsource some or all of their 11 requests to a lawyer or other delegate.

Adrian

**Comment by OR13** [comment]
Here is a hypothetical use case:

https://www.whitehouse.gov/briefing-room/statements-releases/2022/02/26/joint-statement-on-further-restrictive-economic-measures/

```json5
"issuer": [{
  "name": "European Commission",
  "id": "https://press.eu.example/issuers/1"  // example, not an endorsement
},{
  "name": "France",
  "id": "https://press.fr.example/issuers/2"  // example, not an endorsement
},{
  "name": "Germany",
  "id": "did:web:press.de.example"  // example, not an endorsement
},{
  "name": "Italy",
  "id": "did:ethr:..." // example, not an endorsement
},{
  "name": "United Kingdom",
  "id": "did:btcr:..."  // example, not an endorsement
},{
  "name": "Canada",
  "id": "did:indy:..."  // example, not an endorsement
},{
  "name": "United States",
  "id": "did:web:example.gov"  // example, not an endorsement
}]
```

Would the statements about the subject be more or less meaningful with only 1 issuer?

There are lots of scenarios where issuer's might only feel comfortable making statements about a subject together, a quick google search for "joint statements" reveals a lot of potential use cases...

**Comment by OR13** [comment]
Example:

https://v.jsld.org/3cy82JzSKePg2ejfikm3oWhpA1t2XHREWbYW7CMCxvCBMrCJx1bjy9peTgn5q7WrEDMwgAHar3sN1timCQBTJ7Rnymdy6NW5xQJuU4TajAUTTkvFrqs9jgPqamvws9cuyQvi2Kj3TybY86LfjqM9dUS2gsoXcwfYyQ7DgjMERgKgMzFm7m7r7sBGdAQt3BBBfP7aQGt7mCBcrqV99zhuPUAX7Cn23Lutkkg2QrCSZBjvpFR6TZ4Mdv8MCzjjPqyb4V9ggtbsp9hdqQkL667ToM3ApxrgKJ

```
<ddid:example:456> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/ns/credentials/issuer-dependent#Organization> .
<did:example:123> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/ns/credentials/issuer-dependent#Organization> .
_:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/2018/credentials#VerifiableCredential> .
_:c14n0 <https://www.w3.org/2018/credentials#credentialSubject> <did:example:789> .
_:c14n0 <https://www.w3.org/2018/credentials#issuer> <ddid:example:456> .
_:c14n0 <https://www.w3.org/2018/credentials#issuer> <did:example:123> .
_:c14n0 <https://www.w3.org/2018/credentials#validFrom> "2023-05-09T22:25:23.652Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .

```

Suggest closing, its already supported.

**Comment by dlongley** [comment]
That's not compact form -- won't work. Not supported.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-06-06-vcwg-special#section1-5) on 2023-06-06

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.5. Representing Multi Issuer Credentials in the VCDM (issue vc-data-model#932)

_See github issue [vc-data-model#932](https://github.com/w3c/vc-data-model/issues/932)._




**Brent Zundel:** Representing multi-issuer credentials. Question remains, is this something we want to tackle before CR?  

**Orie Steele:** Similar to multiple subject, all representations and securing mechanisms allow it... I think it can be handled post-CR, if at all. If there aren't strong examples in core spec, by default, not a good thing to be doing, don't think we need to do anything w/ this - close issue, if it remains open, someone should create a good example or issue should be closed.  

**Brent Zundel:** proposal to mark as pending close, any other comments on issue?  

> *Orie Steele:* anything is an object if you are brave enough... especially javascript arrays.

**Andres Uribe:** To Orie's point, we do have a lot of examples of real life use cases, but not clear how it works... co-tweets, parents claims about their children, co-signers on loan... what examples do we need for this?  

> *Orie Steele:* I meant a single spec example, with multiple issuers.

> *Orie Steele:* the `@context` allows multiple issuers, and that is what is signed over in both representations.

**Dave Longley:** Agree with most of what Brent said, but maybe disagree w/ having multiple IDs... what spec says today doesn't allow multiple issuers. Don't think it's actually supported today, if people want a change, that needs to happen before CR.  

**Brent Zundel:** does this need to change before CR, any volunteers?  

> *Manu Sporny:* pdl-ASU: Does this involve guardianship, or is that different?

**Brent Zundel:** multi-issuer VCs could be one way, but not the only way.  

> *Orie Steele:* yep, multi issuer and multi subject are both related / required for some scenarios.

> *Orie Steele:* but afaik, nobody is using them (either).

**Joe Andrieu:** I'd be hard pressed to understand how that would work.  

**Manu Sporny:** Just to highlight, I don't think that's an issue, Phil. You can always do multi-issuer by issuing multiple VCs. That may be distasteful in some cases but it works. You can have two VCs claiming a child, and so on.  
… We're covered as far as use cases are concerned, the question is, is that the most ideal form. To be clear, I don't think we need to deal with this in 2.0.  

> *Dave Longley:* pdl-ASU: Very good.

> *Andres Uribe:* +1 for future.

**Brent Zundel:** ok, so suggestion that this is not a VC2.0 thing -- do we mark pending close, or mark future?  

**Joe Andrieu:** Do you want to keep this around, Phil?  

> *Manu Sporny:* pdl-ASU: I just want to make sure we consider this in v3.0... however we keep it around is good.

**Manu Sporny:** I guess the only concern I have with keeping it around is that our issuer tracker ends up collecting a lot of cruft over time. Things that are continuously not addressed. If this is an important enough use case it will be raised again saying they want the spec changed. My preference is to close as many issues as possible if they haven't been worked on.  
… The WG has been operating since 1.0 and 2+ years since then and no one hasn't worked on it. The history will still be there if we close it.  

> *Dave Longley:* pdl-ASU: I will check with some people I know doing guardianship and see if they are ok with handling it.

**Joe Andrieu:** We would accept use cases in the VC use cases document.  

> *Manu Sporny:* pdl-ASU: That's a good suggestion, thanks Joe.

> *Joe Andrieu:* +1.

> *Dave Longley:* +1 to brent.

**Brent Zundel:** So, mark this pending close because there is not consensus to do it. Any objections?  

**Manu Sporny:** +1.  

**Brent Zundel:** ok, marking pending close.  

</details>

**Comment by decentralgabe** [comment]
Closing as there are other ways to address the given use cases.
